objc/objc_runtime_darwin.go

```diff
@@ -148,203 +168,171 @@ func AllocateClassPair(super Class, name string, extraBytes uintptr) Class {
 	return objc_allocateClassPair(super, name, extraBytes)
 }
 
-// Selector is an interface that takes a Go method name
-// and returns the selector equivalent name.
-// If it returns a nil SEL then that method
-// is not added to the class object.
-type Selector interface {
-	Selector(string) SEL
+// MethodDef represents the Go function and the selector that ObjC uses to access that function.
+type MethodDef struct {
+	Cmd SEL
+	Fn  any
 }
 
-// TagFormatError occurs when the parser fails to parse the objc tag in a Selector object
-var TagFormatError = errors.New(`objc tag doesn't match "ClassName : SuperClassName <Protocol, ...>""`)
+// IvarAttrib is the attribute that an ivar has. It affects if and which methods are automatically
+// generated when creating a class with RegisterClass. See [Apple Docs] for an understanding of these attributes.
+// The fields are still accessible using objc.GetIvar and objc.SetIvar regardless of the value of IvarAttrib.
+//
+// Take for example this Objective-C code:
+//
+//	@property (readwrite) float value;
+//
+// In Go, creates functions with these signatures:
+//
+//	func value() float32
+//	func setValue(float32)
```

Comment:
This seems a little confusing. Such these functions are not generated, right?

Reply:
No they aren't exactly created but are equivalently created on the ObjC side

Reply:
Thanks. IIUC, a purego user can access the ivar via `Send` + selectors (`value` and `setValue:`), so I am not sure these Go function sigunatures help users. What about explaining a usage of `Send` and selectors?